### ISSUE TITLE
Should I use different learning rate when using op=Adasum in DistributedOptimizer?

### ISSUE THREAD

**Issue by PANXiao1994** [horovod/horovod#1653]
I want to use [backward_passes_per_step](https://github.com/horovod/horovod/blob/master/horovod/tensorflow/__init__.py#L443-L446) feature to implement gradient accumulation in TensorFlow. And it is obliged to use op=Adasum. However, I suffer from significant performance loss when I use this op, **is this an issue of learning rate?**

The optimizer I use is Adam. I followed the learning rate of paper <Attention is All You Need>. The gradients in one step are simply summed up and the learning rate is adapted according to batch size: 

![image](https://user-images.githubusercontent.com/24204710/72138351-9f291680-33c7-11ea-9377-e94c0f72dd5e.png)

where warmup_steps = 4000 * max(min(25000. / true_batch_tokens, 1.), 0.25).

After switching to Adasum, I didn't modify this learning rate scheme. So is the performance loss due to the learning rate scheme?

**Comment by PANXiao1994** [comment]
I realized that it may be caused by the Algorithm that I use to locally accumulate the gradients within one single GPU:
```
loss_var = tf.get_variable(name="replica_loss",
                                           shape=[],
                                           dtype=tf.float32,
                                           initializer=tf.zeros_initializer,
                                           trainable=False)
slot_variables = _replicate_variables(variables)
zero_variables_op = _zero_variables(slot_variables + [loss_var])
collect_grads_op = _collect_gradients(gradients, slot_variables)
collect_loss_op = tf.assign_add(loss_var, loss)
collect_op = tf.group(collect_loss_op, collect_grads_op, name="collect_op")

## actually I used op=Average within one single GPU
scale = 1.0 / float(update_cycle)
gradients = [scale * (g + s) for (g, s) in zip(gradients, slot_variables)]
loss = scale * (loss + loss_var)
# clip gradients
if self.params["optimizer.clip_gradients"] > 0:
    gradients, _ = tf.clip_by_global_norm(
        gradients, self.params["optimizer.clip_gradients"])
    # grads_and_vars = _clip_gradients(grads_and_vars)
grads_and_vars = list(zip(gradients, variables))
# Create gradient updates.
grad_updates = self._optimizer.apply_gradients(
    grads_and_vars,
    global_step=tf.train.get_global_step(),
    name="train")

# wrap the operations
train_ops = {
    "zeros_op": zero_variables_op,
    "collect_op": collect_op,
    "train_op": grad_updates
}
```

and  during training:

```
try:
    while update_cycle[0] != update_cycle[1]:
        data = train_data.next()
        step_context.session.run(
            train_ops["collect_op"], feed_dict=data["feed_dict"])
        update_cycle[1] += 1
    data = train_data.next()
    update_cycle[1] = 1
    return step_context.run_with_hooks(
        train_ops["train_op"], feed_dict=data["feed_dict"])
except StopIteration:
    eidx[1] += 1
```

Is that issue caused by the inconsistency of "Average" used within single GPU and "Adasum" among GPUs?

**Comment by PANXiao1994** [comment]
@tgaddair

**Comment by tgaddair** [comment]
Hey @PANXiao1994, the learning rate always needs to be scaled when using Horovod, as your batch size will vary with the number of workers you're running with.  For Adasum, the learning rate needs to be scaled by `lr * hvd.local_size()`, while for Average it's `lr * hvd.size()`.

**Comment by Tron-x** [comment]
@PANXiao1994 Hi，I'm interested in large batch size training. Can we talk about this adasum in private firstly? My contact information is zz3223zz@163.com. You can send me your wechat. I have some computing resources. I also want to do this experiment~

**Comment by rosrad** [comment]
hi @tgaddair  and @PANXiao1994 

I have observed the similar trend that the adasum do not convert to a better point
I have done a comparison experiment with 3 differernt setting
with avg, adasum, adasum with lr*hvd.local_sie
the experiment are trained in the same 1x4 gpus
I found that 
1. with the same lr, the adasum do not convert to a better point
2. enlarge the lr by hvd.local_size(), the adasum do not help, but become worse

![image](https://user-images.githubusercontent.com/2343640/72858346-77a54880-3cfb-11ea-9fc2-0683faba2002.png)
